backend/baseclass.pm

```diff
@@ -2,6 +2,46 @@
 # Copyright 2012-2021 SUSE LLC
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+# Helper class to identify file descriptors in error case, see do_capture()
+package NamedIOSelect;
+use Mojo::Base -base;
+use IO::Select;
+use Carp;
+
+sub select {
+    return shift->{select_obj} //= IO::Select->new();
+}
+
+sub names {
+    return shift->{names_hash} //= {};
+}
+
+sub add
+{
+    my ($self, $fd, $name) = @_;
+    my $fd_nr = fileno($fd);
+    if (!defined($name)) {
+        my ($package, $filename, $line) = caller;
+        $name = sprintf('NamedIOSelect::add(%d) called at %s:%d', $fd_nr, $filename, $line);
+    }
```

Comment:
Is this a leftover from debugging?

Reply:
Actually not, If someone is using it the `IO::Select->add()` way (without specify a name), we get descriptive name by using caller.